### PR TITLE
feat: Rocm bumpenvs

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Ticket
https://hpe-aiatscale.atlassian.net/browse/MD-462

## Description
Bumpenvs for rocm
Helm changes for rocm
Release notes

## Test Plan
Determined deploys with k8s.
On an existing k8s deployment, run:
`helm upgrade determined helm/charts/determined/   --values helm/charts/determined/values.yaml`
Then make sure that there is a default value for rocm gpus in master config.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/release-notes/rocm-mi300x.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+:orphan:
+
+**New Features**
+
+-  Added New ROCM 6.1 images with deepspeed for MI300x users. Dev versions of these images can be
```
Comment (tara-det-ai) [inline docs/release-notes/rocm-mi300x.rst:5]:
```suggestion
- Add New ROCM 6.1 images with DeepSpeed for MI300x users. Dev versions of these images can be
```

File: docs/release-notes/rocm-mi300x.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+   found in our dockerhub, under `pytorch-infinityhub-dev
```
Comment (tara-det-ai) [inline docs/release-notes/rocm-mi300x.rst:6]:
```suggestion
found in our Docker Hub, under `pytorch-infinityhub-dev
```

File: docs/release-notes/rocm-mi300x.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+   <https://hub.docker.com/repository/docker/determinedai/pytorch-infinityhub-dev/tags>`__ and
+   `pytorch-infinityhub-hpc-dev
+   <https://hub.docker.com/repository/docker/determinedai/pytorch-infinityhub-hpc-dev/tags>`__.
+   Users can build these images locally based on the dockerfiles found in our `environments
```
Comment (tara-det-ai) [inline docs/release-notes/rocm-mi300x.rst:10]:
```suggestion
Users can build these images locally based on the Dockerfiles found in our `environments
```

**Review by tara-det-ai — APPROVED** [review]
LGTM

File: docs/release-notes/rocm-mi300x.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
+:orphan:
+
+**New Features**
+
+-  Add New ROCM 6.1 images with DeepSpeed for MI300x users. Dev versions of these images can be
+   found in our Docker Hub, under `pytorch-infinityhub-dev
```
Comment (jgongd) [inline docs/release-notes/rocm-mi300x.rst:6]: 🔥

**Review by jgongd — APPROVED** [review]
Modev stamp

**Review by gt2345 — APPROVED** [review]
Web part looks good

File: helm/charts/determined/templates/master-config.yaml [hunk 5]
```diff
@@ -246,28 +246,32 @@ stringData:
...
-      image:
+        {{- else }}
          cpu: {{ .Values.defaultImages.cpuImage | quote }}
+        {{- end }}
+        {{- if .Values.taskContainerDefaults.gpuImage }}
+         gpu: {{ .Values.taskContainerDefaults.gpuImage | quote }}
+        {{- else }}
          gpu: {{ .Values.defaultImages.gpuImage | quote }}
-      {{- if or .Values.taskContainerDefaults.cpuImage .Values.taskContainerDefaults.gpuImage }}
-        {{ required "A valid .Values.taskContainerDefaults.cpuImage entry is required if setting .Values.taskContainerDefaults.gpuImage!" .Values.taskContainerDefaults.cpuImage }}
```
Comment (NicholasBlaskey) [inline helm/charts/determined/templates/master-config.yaml:258]: we are removing this required check meaning you no longer need to set both a gpu / cpu image right?

I think someones cluster will likely not work if they don't have both a cpu and gpu image specified

Comment (MikhailKardash) [reply helm/charts/determined/templates/master-config.yaml:258]: Yeah. In this rewrite, there is always a `cpu` and `gpu` image. Whichever one isn't set will be inferred from `.Values.defaultImages`. (same goes for `rocm`).

**Review by NicholasBlaskey — APPROVED** [review]

**Pull request merged**